### PR TITLE
Initial implementation of single-stratum vaccination

### PR DESCRIPTION
**Pull request by pratikunterwegs**
This PR is an initial implementation of vaccination.

- Total of two vaccine groups/strata - unvaccinated and vaccinated;
- Vaccination has a uniform rate and start time across age groups;
- Vaccination starts when a response threshold (time or hospital capacity) is reached, and is active in the 'no response' scenario as well;
- Vaccination does not end;
- Flows between vaccination groups/strata only affect susceptible or recovered individuals;
- Vaccination reduced susceptibility to infection by 50% (arbitrary);
- Vaccine-derived protection wanes over 270 days (arbitrary); this is in addition to infection-derived protection - so recovered and vaccinated individuals can flow back into the recovered and unvaccinated compartment.

## Implementation details

- Vaccination rate is passed as an argument `vaccination_rate` to `daedalus()`;
- Vaccination strata are represented along a newly added 4th dimension of the state array;
- All ODE code accounts for addition of vaccination groups;
- Vaccination rate is scaled by number of vaccination-eligible individuals remaining, so that vaccination rate more accurately reflects doses-per-day (expressed in population proportions);
- Vaccination rate updating is highly dependent on the assumption of uniform rate and start time;

## To do:

Issues remaining which need discussion:

- Vaccination-to-response mapping is worked out, i.e., rates, activation, etc.;
- Vaccination only for oldest age group?;
- Vaccination up to the uptake limit terminates active response? **This will need some thought!**;
- Should vaccination be limited by number of doses available?

### CODE REVIEW

File: R/prepare_output.R [hunk 1]
```diff
@@ -15,35 +15,61 @@
...
+  )
+  econ_sector <- rep(
+    econ_sector, N_VACCINE_STRATA
   )
+  data$Var4 <- econ_sector
 
-  # data values
-  values <- c(output[, setdiff(colnames(output), "time")])
+  # vaccine group labels
+  data$Var5 <- rep(
```
Comment (richfitz) [inline R/prepare_output.R:64]: Why assigning here to `Var5` to replace with `vaccine_group` later? Why not `data$vaccine_group <- ...` (and elsewhere here)

**Review by richfitz — CHANGES_REQUESTED** [review]

File: R/prepare_output.R (hunk 1 above)
Comment (pratikunterwegs) [reply R/prepare_output.R:64]: No good reason really - that way I'd create new columns and subset out the ones we don't need. This saves a subset and adds a renaming operation. No preference either way.

**Comment by pratikunterwegs** [comment]
Spotted a little issue with the closure activation and termination functions where the `state` hasn't been updated to reflect the newly added fourth dimension - will fix and push.

**Comment by pratikunterwegs** [comment]
Fixed:
- a conditional in `scale_nu()` - $\nu$ is 0.0 if not finite;
- updated dims in state used in rootfinding-event functions;
- a failing test where vaccination led to increased deaths.

File: R/vaccination.R
Comment (richfitz) [inline R/vaccination.R:1]: sorry about that!

**Review by richfitz — APPROVED** [review]

**Pull request merged**